sklearn/linear_model/stochastic_gradient.py

```diff
@@ -1029,7 +1019,7 @@ def _fit_regressor(self, X, y, alpha, C, loss, learning_rate,
         penalty_type = self._get_penalty_type(self.penalty)
         learning_rate_type = self._get_learning_rate_type(learning_rate)
 
-        if self.t_ is None:
+        if getattr(self, "t_", None) is None:
```

Comment:
why not `not hasattr(self, 't_')` instead? If this is to support performing a warm started fit on a unpickled previously trained with a past version of scikit-learn, we should add a comment to make it explicit.